utils.py

```diff
@@ -287,9 +300,21 @@ def get_runmode():
     else:
         return 'script'
 
-def install_packages(*packages):
-    command = [config.SUPERUSER_COMMAND, config.PACKAGE_MANAGER_COMMAND] + list(packages)
-    subprocess.run(command, check=True)
+def query_packages(packages):
+    missing_packages = [ ]
+    
+    for p in packages:
+        command = f"{config.SUPERUSER_COMMAND} {config.PACKAGE_MANAGER_COMMAND_QUERY}{p}"
+        result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, text=True)
+        if result.returncode != 0:
+            missing_packages.append(p)
```

Comment:
I don't know about other systems, but ubuntu doesn't need `sudo` to query package status. I think this would require an unnecessary password prompt if the user has all the dependencies already installed.

Reply:
The same is true for Arch, and I would imagine the others.